pkg/reconciler/workload/placement/placement_controller.go

```diff
@@ -250,50 +250,35 @@ func (c *controller) enqueueAPIBinding(obj interface{}) {
 		return
 	}
 
-	logger := logging.WithObject(logging.WithReconciler(klog.Background(), ControllerName), obj.(*apisv1alpha1.APIBinding))
+	logger = logger.WithValues(logging.FromPrefix("apiBindingReason", obj.(*apisv1alpha1.APIBinding))...)
+
 	for _, placement := range placements {
-		c.enqueuePlacement(placement, logger, " because of APIBinding")
+		c.enqueuePlacement(placement, logger)
 	}
 }
 
-func (c *controller) enqueueSyncTarget(obj interface{}) {
-	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
+func (c *controller) enqueueSyncTarget(obj interface{}, logger logr.Logger) {
```

Comment:
this is very noisy. We should rethink whether we really need to enqueue half of the world.

1. What is the exact reason we have to enqueue a sync target?
2. What can we do to get the placements more directly?

Reply:
Goal: only sync the placement that actually schedules to that SyncTarget.

Reply:
ideally we should only enqueue location here. The missing part is  we do not know what APIs are compatible in the location. So if a synctarget is updated with "api compaible/non-comptaible", we have to select the clusters here.

Reply:
> So if a synctarget is updated with "api compaible/non-comptaible"

For that we could have a special case, couldn't we?

Reply:
Also, can we queue only those locations that are affected, e.g. that include the SyncTarget? If you remove a label from a SyncTarget, we could also see that and queue those locations where this SyncTarget was removed from.

Reply:
Nvmd:

On the other hand, the new change does not really cause more noise, but it will list all locations and from there get all placements that reference these locations. So there is hope that these are not all placements, vs. the old code that queued all placements.